core/src/main/java/jenkins/model/ArtifactManager.java

```diff
@@ -60,11 +60,12 @@
      * @param launcher a launcher to use if external processes need to be forked
      * @param listener a way to print messages about progress or problems
      * @param artifacts map from paths in the archive area to paths relative to {@code workspace} (all paths {@code /}-separated)
+     * @param targetDirectory the target parent directory
      * @throws IOException if transfer or copying failed in any way
      * @throws InterruptedException if transfer was interrupted
      * @see ArtifactArchiver#perform(Run, FilePath, Launcher, TaskListener)
      */
-    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String,String> artifacts) throws IOException, InterruptedException;
+    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String,String> artifacts, String targetDirectory) throws IOException, InterruptedException;
```

Comment:
Change breaking API compatibility.

Reply:
@daniel-beck I agree. Should I add another method signature or find antoher way that avoid modifying the API?

Reply:
Yes. Add another method with more arguments, and invoke that one from this one with whatever is the "default" argument.